File: src/askem_beaker/contexts/dataset/procedures/julia/df_save_as.jl
using Dates

_temp_csv_file = tempname()
CSV.write(_temp_csv_file, {{ var_name|default("df") }}, writeheader=true)
_filesize = stat(_temp_csv_file).size
_upload_response = HTTP.put("{{data_url}}", ["content-length" => _filesize], open(_temp_csv_file, "r"))

if _upload_response.status != 200
    error("Error uploading dataframe: $(String(_upload_response.body))")
end

# Cleanup
rm(_temp_csv_file)

File: src/askem_beaker/contexts/dataset/procedures/rlang/df_save_as.r
import requests
import tempfile

# Saving as a temporary file instead of a buffer to save memory
with tempfile.TemporaryFile() as temp_csv_file:
    {{ var_name|default("df") }}.to_csv(temp_csv_file, index=False, header=True)
    temp_csv_file.seek(0)
    upload_response = requests.put('{{data_url}}', data=temp_csv_file)
if upload_response.status_code != 200:
    raise Exception(f"Error uploading dataframe: {upload_response.content}")
